Dockerfile

```diff
@@ -39,7 +39,7 @@ ENV GPU_NODE_MIN_SIZE="0"
 ENV GPU_PER_NODE="1"
 
 # gcloud config
-ENV KUBERNETES_VERSION="latest"
+ENV KUBERNETES_VERSION="1.14"
```

Comment:
Kubernetes has a really fast (compared to something like Ubuntu, which I'm familiar with) release schedule:
https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions

The most concerning part is that a given minor release ends up only being supported for 9 months total. In the Kubernetes docs, they don't even talk about anything below 1.16 anymore.

Using a known version is desirable but, if we're going to pin it now, maybe we should try to pin to the newest version possible.

Reply:
The newest Kubernetes version available on GKE (1.16.8) is incompatible with our helm installation (#345).  GKE is currently using 1.14 as the default version.

We can always have a minor release ourselves if we need to upgrade the default value for `KUBERNETES_VERSION`.

Reply:
Looking at GKE's release notes, it looks like they only support a minor version of Kubernetes for a year before they start auto-upgrading clusters using that minor to the next minor:
https://cloud.google.com/kubernetes-engine/docs/release-notes
https://cloud.google.com/kubernetes-engine/docs/release-notes-archive

It looks like 1.14 is next on the chopping block, since they started auto-upgrading 1.13 to 1.14 in late March:
https://cloud.google.com/kubernetes-engine/docs/release-notes#scheduled_automatic_upgrades_4

Kubernetes' rapid release cadence and short support lifetimes for minor versions make it difficult to keep up with. GKE makes things a little easier by apparently lengthening the minor version lifetime from 9 months to ~1 year, but that's still not a very long time.

There's not a clear best solution here to me, but I think we should at least discuss this further, since pinning to 1.14 is likely going to lead to auto-upgrading within the next three months. I'm worried about auto-upgrading happening in `master`, especially since we're trying to support single-zone clusters again and auto-upgrading was our initial impetus for moving to multi-zone clusters.

Reply:
My understanding is that there are 4 valid values for `KUBERNETES_VERSION`: 
- `1.14`
- `1.15`
- `1.16`
- `latest`

`latest` and `1.16` are both currently incompatible with our cluster, which leaves 1.14 and 1.15. I think we should make this release on 1.14, since we have primarily tested it against this version, and the next `stable` change can upgrade to 1.15. Do you think we should move this to 1.15 now instead?

Reply:
Do we support single-zone clusters in this release? If so, are they the default?

If they're not the default, then I say let's just document that they may not work, with the expectation that they'll probably start failing in a few months and try to move to a higher version before then.

If they are the default, then I think we really should move to the newest version we can. Looking at the release notes, it looks like 1.17 became available a few weeks ago. It's unlikely, but maybe it'll work out of the box?

Reply:
Single zone clusters are enabled. The user can select Multizone or one of the available zones; the default is Multizone, but its a menu so they can easily select a single zone. Maybe we can open an issue that describes how we have seen single-zone clusters fail deployments due to upgrading masters.

"latest" is set to 1.16.8 in static releases. 1.17 is available in the Rapid release channel, but:

> Rapid channel is offered on an early access basis for customers who want to test new releases before they are qualified for production use or general availability. Versions available in the Rapid Channel may be subject to unresolved issues with no known workaround and are not for use with production workloads or subject to any SLAs.

I think the next step to upgrading the version used is to fix the `helm` error with `1.16.8`. However, my concern about changing the version before the 1.2.0 release is that we have done all our stress-testing on the 1.14 release.

Reply:
Ok, that all makes sense. (I wasn't aware that the Rapid channel wasn't meant for production clusters. That's good to know, even though it's a little unfortunate, since it means that the support lifetime for a GKE version that can be used in production is less than a year.)

For the menu, if multi-zone clusters are the default, then it's probably sufficient to warn people about the behavior we've seen, if they select a single-zone cluster. Is there any way we can get that warning into `master` with this merge? I can code it up and open a PR into this branch, if you want.

For the future, we should try to upgrade from 1.14 within the next couple months, since the auto-upgrades are likely to break single-zone clusters. I can write an issue for this, if one doesn't exist yet.

Reply:
Would an issue about 1.14 and single-zone clusters be a sufficient warning? Or do you think we should also include a warning message on the "Select your zone" screen?

Anecdotally, I did not have any issues deploying single-zone clusters when I was testing the single-zone PR.

Reply:
I am wary of the warning as it is now in #352. I definitely understand the concerns re: single-zone clusters, but I think that is better served in an issue as you laid out @dylanbannon. I believe an issue like that would be sufficient for now, as we are in active development and will be for the life of this platform (where no doubt many other external version issues will require attention).

Reply:
Just to summarize, #352 resolves these issues by warning users using KUBERNETES version 1.14 and the date is after the expected retirement date of 1.14.

Reply:
@MekWarrior I agree. I think #352 is a great fix for this. Upgrading has to happen eventually, but it's probably best to put it off, especially since we benchmarked our publication data on this version.

Once #352 gets merged into `stable`, everything looks good to me.